proctl/breakpoints_linux_amd64.go

```diff
@@ -50,6 +51,15 @@ func PtracePokeUser(tid int, off, addr uintptr) error {
 	return nil
 }
 
+func PtracePeekUser(tid int, off uintptr) (uintptr, error) {
+	var x uintptr // XXX: this should not be necessary
+	ret, _, err := syscall.Syscall6(syscall.SYS_PTRACE, syscall.PTRACE_PEEKUSR, uintptr(tid), uintptr(off), uintptr(unsafe.Pointer(&x)), 0, 0)
```

Comment:
`ret` will never be anything of interest to us here. The actual value from this ptrace request will be stored in `x`. The ptrace API is to provide an address and data, and for reads (PEEK) the data stored in the requested address will be placed in the data argument, which here would be `x`.

`ret` will be 0 every time, whereas `x` will hold to value of interest.